orchagent/policerorch.cpp

```diff
@@ -123,119 +141,396 @@ void PolicerOrch::doTask(Consumer &consumer)
+            continue; 
+        }
+
+        /*Policer Name: _<interface_name>_<storm_type>*/
+        const auto storm_policer_name = "_"+interface_name+"_"+storm_type;
+
+        if (op == SET_COMMAND)
+        {
+            // Mark the opeartion as an 'update', if the policer exists.
+            bool update = m_syncdPolicers.find(storm_policer_name) != m_syncdPolicers.end();
+            vector <sai_attribute_t> attrs;
+            bool cir = false;
+            sai_attribute_t attr;
+
+            /*Meter type hardcoded to BYTES*/
+            attr.id = SAI_POLICER_ATTR_METER_TYPE;
+            attr.value.s32 = (sai_meter_type_t) meter_type_map.at("BYTES");
+            attrs.push_back(attr);
+
+            /*Policer mode hardcoded to STORM_CONTROL*/
+            attr.id = SAI_POLICER_ATTR_MODE;
+            attr.value.s32 = (sai_policer_mode_t) policer_mode_map.at("STORM_CONTROL");
+            attrs.push_back(attr);
+
+            /*Red Packet Action hardcoded to DROP*/
+            attr.id = SAI_POLICER_ATTR_RED_PACKET_ACTION;
+            attr.value.s32 = packet_action_map.at("DROP");
+            attrs.push_back(attr);
+
+            for (auto i = kfvFieldsValues(tuple).begin();
+                    i != kfvFieldsValues(tuple).end(); ++i)
+            {
+                auto field = to_upper(fvField(*i));
+                auto value = to_upper(fvValue(*i));
+
+                /*BPS value is used as CIR*/
+                if (field == storm_control_kbps)
                 {
                     attr.id = SAI_POLICER_ATTR_CIR;
-                    attr.value.u64 = stoul(value);
-                }
-                else if (field == pbs_field)
-                {
-                    attr.id = SAI_POLICER_ATTR_PBS;
-                    attr.value.u64 = stoul(value);
-                }
-                else if (field == pir_field)
-                {
-                    attr.id = SAI_POLICER_ATTR_PIR;
-                    attr.value.u64 = stoul(value);
-                }
-                else if (field == red_packet_action_field)
-                {
-                    attr.id = SAI_POLICER_ATTR_RED_PACKET_ACTION;
-                    attr.value.s32 = packet_action_map.at(value);
-                }
-                else if (field == green_packet_action_field)
-                {
-                    attr.id = SAI_POLICER_ATTR_GREEN_PACKET_ACTION;
-                    attr.value.s32 = packet_action_map.at(value);
-                }
-                else if (field == yellow_packet_action_field)
-                {
-                    attr.id = SAI_POLICER_ATTR_YELLOW_PACKET_ACTION;
-                    attr.value.s32 = packet_action_map.at(value);
+                    /*convert kbps to bps*/
+                    attr.value.u64 = (stoul(value)*1000/8);
+                    cir = true;
+                    attrs.push_back(attr);
+                    SWSS_LOG_INFO("CIR %s",value.c_str());
                 }
                 else
                 {
```

Comment:
since the default behavior is "enabled" didn't implement the field here. Can be introduced in a scenario where the CONFIG_DB table contains the configuration but it isn't programmed in the hardware possibly because of hardware limitation.